src/convert-doi-links-to-pub-id-elements.xsl

```diff
@@ -10,10 +10,10 @@
     </xsl:copy>
   </xsl:template>
 
-  <xsl:template match="ext-link[@ext-link-type='uri'][starts-with(lower-case(@xlink:href), 'https://doi.org/')]">
+  <xsl:template match="ref//ext-link[@ext-link-type='uri'][matches(lower-case(@xlink:href),'^https?://(dx\.)?doi\.org/')]">
     <xsl:element name="pub-id">
       <xsl:attribute name="pub-id-type">doi</xsl:attribute>
-      <xsl:value-of select="substring(@xlink:href, string-length('https://doi.org/')+1)"/>
+      <xsl:value-of select="substring(@xlink:href, (string-length(@xlink:href) - string-length(substring-after(lower-case(@xlink:href),'doi.org/')) + 1))"/>
     </xsl:element>
```

Comment:
Suggest we just go with `substring-after(lower-case(@xlink:href),'doi.org/')` which should work in all cases that match the regular expression.